webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -311,143 +317,138 @@ class PolicyWriterServiceImpl(
       // Clearing cache
-      _                 = fillTemplates.clearCache
-      readTemplateTime2 = System.currentTimeMillis
-      readTemplateDur   = readTemplateTime2 - readTemplateTime1
-      _                 = policyLogger.debug(s"Paths computed and templates read in ${readTemplateDur} ms")
+      _                    <- IOResult.effect(fillTemplates.clearCache)
+      readTemplateTime2    <- currentTimeMillis
+      _                    <- timingLogger.debug(s"Paths computed and templates read in ${readTemplateTime2 - readTemplateTime1} ms")
 
       //////////
       // nothing agent specific before that
       //////////
 
-      preparedPromises <- parrallelSequence(configAndPaths) { case agentNodeConfig =>
-                            val nodeConfigId = versions(agentNodeConfig.config.nodeInfo.id)
-                            prepareTemplate.prepareTemplateForAgentNodeConfiguration(agentNodeConfig, nodeConfigId, rootNodeId, templates, allNodeConfigs, Policy.TAG_OF_RUDDER_ID, globalPolicyMode, generationTime) ?~!
-                            s"Error when calculating configuration for node '${agentNodeConfig.config.nodeInfo.hostname}' (${agentNodeConfig.config.nodeInfo.id.value})"
-                         }
-      preparedPromisesTime = System.currentTimeMillis
-      preparedPromisesDur  = preparedPromisesTime - readTemplateTime2
-      _                    = policyLogger.debug(s"Promises prepared in ${preparedPromisesDur} ms")
-
-      promiseWritten   <- parrallelSequence(preparedPromises) { prepared =>
-                            (for {
-                              _ <- writePromises(prepared.paths, prepared.agentNodeProps.agentType, prepared.preparedTechniques, resources)
-                              _ <- writeAllAgentSpecificFiles.write(prepared) ?~! s"Error with node '${prepared.paths.nodeId.value}'"
-                              _ <- writeDirectiveCsv(prepared.paths, prepared.policies, globalPolicyMode)
-                              _ <- writeSystemVarJson(prepared.paths, prepared.systemVariables)
-                            } yield {
-                              "OK"
-                            }) ?~! s"Error when writing configuration for node '${prepared.paths.nodeId.value}'"
-                          }
-      promiseWrittenTime = System.currentTimeMillis
-      promiseWrittenDur  = promiseWrittenTime - preparedPromisesTime
-      _                  = policyLogger.debug(s"Policies written in ${promiseWrittenDur} ms")
+      preparedPromises     <- parrallelSequence(configAndPaths) { case agentNodeConfig =>
+                                val nodeConfigId = versions(agentNodeConfig.config.nodeInfo.id)
+                                prepareTemplate.prepareTemplateForAgentNodeConfiguration(agentNodeConfig, nodeConfigId, rootNodeId, templates, allNodeConfigs, Policy.TAG_OF_RUDDER_ID, globalPolicyMode, generationTime, maxParallelism).chainError(
```

Comment:
i'm a bit confused by the expected gain of a paralelised computation within a parrallelSequence